run-workloads.sh

```
@@ -487,6 +487,33 @@ run_single_test() {
+    local _overridden=()
+    while IFS= read -r _line; do
+        [[ "$_line" =~ ^[[:space:]]*# ]] && continue
+        [[ -z "$_line" ]] && continue
+        if [[ "$_line" =~ ^([a-zA-Z_][a-zA-Z0-9_]*):(.*)$ ]]; then
+            local _key="${BASH_REMATCH[1]}"
+            local _orig="${BASH_REMATCH[2]}"
+            [[ "$_key" == "PROM" || "$_key" == "PROM_TOKEN" || "$_key" == "esServer" ]] && continue
+            if [[ -n "${!_key+x}" && -n "${!_key}" ]]; then
+                local _val="${!_key}"
+                if [[ "$_orig" == *'"'* || "$_orig" == *"'"* ]]; then
+                    sed -i "s|^${_key}:.*|${_key}: \"${_val}\"|" "$temp_vars"
```

Shellcheck (0.11.0)

[info] Make sure not to read and write the same file in the same pipeline.

(SC2094)